src/posix/platform/radio.cpp

```diff
@@ -147,6 +118,34 @@ void platformRadioInit(otUrl *aRadioUrl)
         VerifyOrDie(INT8_MIN <= ccaThreshold && ccaThreshold <= INT8_MAX, OT_EXIT_INVALID_ARGUMENTS);
         SuccessOrDie(sRadioSpinel.SetCcaEnergyDetectThreshold(static_cast<int8_t>(ccaThreshold)));
     }
+
+#if OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE
+    maxPowerTable = radioUrl.GetValue("max-power-table");
+    if (maxPowerTable != nullptr)
+    {
+        constexpr int8_t kPowerDefault = 30; // Default power 1 watt (30 dBm).
+        const char *     str           = nullptr;
+        uint8_t          channel       = ot::Radio::kChannelMin;
+        int8_t           power         = kPowerDefault;
+
+        for (str = strtok(const_cast<char *>(maxPowerTable), ","); str != nullptr && channel <= ot::Radio::kChannelMax;
+             str = strtok(nullptr, ","))
+        {
+            power = static_cast<int8_t>(strtol(str, nullptr, 0));
+            SuccessOrDie(sRadioSpinel.SetChannelMaxTransmitPower(channel, power));
```

Comment:
I notcied in `RestoreProperties()`, there is check for if the result is `OT_ERROR_NOT_FOUND` (i.e. to protect when RCP is older version and does not support the newly added spinel property for MAX_POWER).

But here seems like here we use `SuccessOrDie` and if RCP does not support the property we would exit.

- Either we need to update this code and allow host side code to work with older RCP version, 
- Or we should update the `SPINEL_MIN_HOST_SUPPORTED_RCP_API_VERSION` along with `SPINEL_RCP_API_VERSION` to ensure/verify correct compatibility.

Reply:
I didn't add the check since it's protected in a feature macro but it looks reasonable to tolerate older RCP versions.